examples/poisson/causal_test_setup.py

```diff
@@ -0,0 +1,197 @@
+        :return: The conditional average treatment effect and the 95% Wald confidence intervals.
+        """
+        assert self.effect_modifiers, f"Must have at least one effect modifier to compute CATE - {self.effect_modifiers}."
+        x = pd.DataFrame()
+        x[self.treatment[0]] = [self.treatment_values, self.control_values]
+        x['Intercept'] = 1
+        for k, v in self.effect_modifiers.items():
+            self.adjustment_set.add(k)
+            x[k] = v
+        if hasattr(self, "product_terms"):
+            for a, b in self.product_terms:
+                x[f"{a}*{b}"] = x[a] * x[b]
+
+        x.drop(['width', 'height'], axis=1, inplace=True)
+        self.adjustment_set = {"width*intensity", "height*intensity"}
+
+        print(x)
+        print(self.adjustment_set)
+
+        model = self._run_linear_regression()
+        print(model.summary())
+        y = model.predict(x)
+        treatment_outcome = y.iloc[0]
+        control_outcome = y.iloc[1]
+
+        return treatment_outcome - control_outcome, None
+
+
+class PoissonWidthHeight(CausalTestOutcome):
+    """An extension of TestOutcome representing that the expected causal effect should be positive."""
+
+    def __init__(self, tolerance=0.5):
+        self.tolerance = tolerance
+        self.i2c = None
+
+    def apply(self, res: CausalTestResult) -> bool:
+        # TODO: confidence intervals?
+        print("=== APPLYING ===")
+        print("effect_modifier_configuration", res.effect_modifier_configuration)
+        effect_modifier_configuration = {k.name: v for k, v in res.effect_modifier_configuration.items()}
+        c = res.treatment_value - res.control_value
+        i = effect_modifier_configuration['intensity']
+        self.i2c = i * 2 * c
+        print("2ic:", f"2*{i}*{c}={self.i2c}")
+        print("ate:", res.ate)
+        return np.isclose(res.ate, self.i2c, atol=self.tolerance)
+
+    def __str__(self):
+        if self.i2c is None:
+            return f"PoissonWidthHeight±{self.tolerance}"
+        return f"PoissonWidthHeight:{self.i2c}±{self.tolerance}"
+
+
+def populate_width_height(data):
+    data['width_plus_height'] = data['width'] + data['height']
+
+
+def populate_num_lines_unit(data):
+    area = data['width'] * data['height']
+    data['num_lines_unit'] = data['num_lines_abs'] / area
+
+
+def populate_num_shapes_unit(data):
+    area = data['width'] * data['height']
+    data['num_shapes_unit'] = data['num_shapes_abs'] / area
+
+
+def get_args() -> argparse.Namespace:
+    """ Command-line arguments
+
+    :return: parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="A script for parsing json config files for the Causal Testing Framework")
+    parser.add_argument("-f", help="if included, the script will stop if a test fails",
+                        action="store_true")
+    parser.add_argument("--directory_path", help="path to the json file containing the causal tests",
```

Comment:
I'm not sure the default will work here, as it's supposed to be setting a folder / directory, not a specific file. And the filenames are hardcoded elsewhere (in `causal_testing/json_front/json_class.py`). Maybe get rid of the CLI arg and look for files in the same folder as the script for now? Or get rid of the default value? We may want to look at this again after merging, depending on expected user needs around file locations and naming. I would have thought that, at least, not everyone would want to call their data `data.csv` which I think is the current requirement.

Reply:
You're right. I think now that the door has been opened in terms of the user writing more python code, the simplest approach is expect the user to specify the paths in `causal_test_setup.py`. This way the file names and locations are completely flexible and up to the user